src/features/pythonApi.ts

```diff
@@ -216,11 +221,75 @@ class PythonEnvironmentApiImpl implements PythonEnvironmentApi {
         }
         return new Disposable(() => disposables.forEach((d) => d.dispose()));
     }
-    installPackages(context: PythonEnvironment, packages: string[], options: PackageInstallOptions): Promise<void> {
+    async installPackages(
+        context: PythonEnvironment,
+        packages: string[],
+        options: PackageInstallOptions,
+    ): Promise<void> {
+        // get the package manager to exit if not found
         const manager = this.envManagers.getPackageManager(context);
         if (!manager) {
             return Promise.reject(new Error('No package manager found'));
         }
+        // what does it mean to return, will we tell the calling extension about it?
+        //check to see if pkg was already installed?
+        const callingExtension = getCallingExtension();
+        traceInfo(`Python API: Installing packages for extension: ${callingExtension}`);
+        const config = getConfiguration('python-envs');
+        let extPkgTrustConfig: SettingsPackageTrust | undefined =
+            config.get<SettingsPackageTrust>('extensionPackageTrust');
+        let callingExtensionTrustLevel;
+        let isConfigured = true;
+        if (extPkgTrustConfig === undefined) {
+            // TODO:s THIS DOESN'T WORK
+            // no package trust config, default to alwaysAsk
+            callingExtensionTrustLevel = 'alwaysAsk';
+            isConfigured = false;
+        } else {
+            // check for package trust settings
+            callingExtensionTrustLevel = extPkgTrustConfig[callingExtension];
+            if (callingExtensionTrustLevel === undefined) {
+                // no specific package trust settings, checking wildcard in config
+                callingExtensionTrustLevel = extPkgTrustConfig['*'];
+                if (callingExtensionTrustLevel === undefined) {
+                    // no wildcard in config, default to alwaysAsk
+                    callingExtensionTrustLevel = 'alwaysAsk';
+                    isConfigured = false;
+                }
+            }
+        }
+        traceInfo(`package trust settings for ${callingExtension} is ${callingExtensionTrustLevel}`);
+
+        if (!isConfigured) {
+            // calling extension has no config, user has no wildcard setup
+            // prompt user to "alwaysAsk" or "alwaysAllow"
+            const selectedOption = await promptForInstallPermissions(callingExtension, packages.join(', '));
+            if (selectedOption === 'cancel') {
+                // user cancelled the prompt, exit
+                window.showErrorMessage(`Package installation of ${packages.join(', ')} was canceled by the user.`);
+                return Promise.resolve();
```

Comment:
It seems like the only two options for what this function returns are resolving it or having it error. When a user cancels the install flow, the error seems to noisy but it is probably necessary to communicate to the calling extension the install didn't happen. Did this come to mind already / thoughts?

Reply:
We should surface the error when a package manager does is not allowed to install. I think it is ok for this error to go to the extension once. This will be useful to surface the UI to allow installs.

Reply:
like one of these as the UI?
<img width="588" alt="Screenshot 2025-02-18 at 2 46 06 PM" src="https://github.com/user-attachments/assets/53d022f7-664a-466b-a05d-71a13593ed72" />
<img width="1718" alt="Screenshot 2025-02-18 at 2 45 40 PM" src="https://github.com/user-attachments/assets/563c5e3c-8f51-4dd1-a661-6f73db46ef77" />

Reply:
so the full screen modal pop up is the better one between the two?

Reply:
this is fixed- it will pass the exception to the extension that called the install and it is up to that extension to handle the error as they see fit